query/src/main/scala/filodb/query/LogicalPlan.scala

```diff
@@ -10,6 +10,13 @@ sealed trait LogicalPlan {
     * It is false for RawSeriesLikePlan, MetadataQueryPlan, RawChunkMeta, ScalarTimeBasedPlan and ScalarFixedDoublePlan
     */
   def isRoutable: Boolean = true
+  def replaceFilters(filters: Seq[ColumnFilter]): LogicalPlan = {
```

Comment:
Some scaladoc on the purpose and contract would help people writing more implementations